src/state/reducers/search.js

```diff
@@ -11,44 +11,60 @@ export const searchesReducer = (state = {}, action) => {
     case ActionTypes.REQUEST_SEARCH:
       return {
         ...state,
-        [action.targetId]: {
-          ...state[action.targetId],
+        [action.windowId]: {
+          ...state[action.windowId],
           [action.companionWindowId]: {
-            isFetching: true,
+            ...(state[action.windowId] || {})[action.companionWindowId],
+            data: {
```

Comment:
I'm noticing that the state change here (while useful for pagination) tanks Mirador when multiple queries are made within the same companion window.

Not only does the memory increase (and hold on until companion window is removed), also the search hits hang on indefinitely and are not updated.

![memory](https://user-images.githubusercontent.com/1656824/59725307-72729f00-91ea-11e9-8877-1274695a39e7.gif)

The mirador instance has crashed a few times on me.